src/tz390.java

```diff
@@ -7684,4 +7688,27 @@ public boolean check_java_version(){
 		}
 		return true;
 }
+public String getVersion() {
+
+    try (InputStream input = getClass().getClassLoader().getResourceAsStream("z390.properties")) {
+
+            Properties prop = new Properties();
+
+            if (input == null) {
+                System.out.println("Sorry, unable to find z390.properties");
+                return null;
+            }
+
+            //load a properties file from class path, inside static method
+            prop.load(input);
+
+            //get the property value and print it out
+            return prop.getProperty("version");
+
+        } catch (IOException ex) {
+            ex.printStackTrace();
+            return null;
+        }
+
+}
 }
```

Comment:
You might want to modify the getVersion() method to add a finally clause to the try-catch so that the input stream is closed. This will require that you only have one return statement at the end. See

https://crunchify.com/java-properties-file-how-to-read-config-properties-values-in-java/

Here's one possibility for modified getVersion() based on the link. Line numbers are the above line numbers.
1.  Above 7693 (before the try) add
String result = null;

2. Replace 7697 -- 7706 with
if (inpt != null) {
    prop.load(input);
    result = prop.getProperty("version');
} else {
    System.out.println("Sorry, unable to find z390.properties/");
}

3. Delete 7710 (the return null; in the catch block)

4. Modify 7711 (the closing brace for the catch block) and add
} finally {
    input.close();
}. 

5. Add "return result;" above 7713.

Reply:
See deccb07 for changes made related to the above comments.

The try (resource) will automatically close the resource so no need for the finally.
see https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html for the syntax and https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html to confirm that InputStream object is autoclosable.
I am setting the version as "Uncontrolled" if the version cannot be found as null is not a good message.
I am also on the fence about displaying the messages -- the uncontrolled tells the story. Really need to get the message at compile time rather than run time.. will have a think about this.